pkg/controllers/state/cluster.go

```diff
@@ -106,8 +110,8 @@ func (c *Cluster) Synced(ctx context.Context) (synced bool) {
 	defer func() {
 		ClusterStateSynced.Set(lo.Ternary[float64](synced, 1, 0), nil)
 	}()
-	nodeClaimList := &v1.NodeClaimList{}
-	if err := c.kubeClient.List(ctx, nodeClaimList); err != nil {
+	nodeClaims, err := nodeclaim.List(ctx, c.kubeClient, nodeclaim.WithManagedFilter(c.cloudProvider))
```

Comment:
Hmmm -- this is interesting -- so we only care that our NodeClaims are synced -- it's an interesting consideration if we should consider NodeClaims that we don't directly manage or not